File: core/src/main/java/com/ustadmobile/core/view/ReportAttendanceGroupedByThresholdsView.java
package com.ustadmobile.core.view;


import com.ustadmobile.core.db.dao.ClazzLogAttendanceRecordDao.AttendanceResultGroupedByAgeAndThreshold;

import java.util.LinkedHashMap;
import java.util.List;

/**
 * ReportNumberOfDaysClassesOpen Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
public interface ReportAttendanceGroupedByThresholdsView extends UstadView {

    String VIEW_NAME = "ReportAttendanceGroupedByThresholds";


    void updateTables(LinkedHashMap<String, List<AttendanceResultGroupedByAgeAndThreshold>> dataMaps);

    /**
     * For Android: closes the activity.
     */
    void finish();

}

File: core/src/commonMain/kotlin/com/ustadmobile/staging/core/view/ReportAttendanceGroupedByThresholdsView.kt
package com.ustadmobile.core.view

import com.ustadmobile.lib.db.entities.AttendanceResultGroupedByAgeAndThreshold


/**
 * ReportNumberOfDaysClassesOpen Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
interface ReportAttendanceGroupedByThresholdsView : UstadView {


    fun updateTables(dataMaps: LinkedHashMap<String, List<AttendanceResultGroupedByAgeAndThreshold>>)

    /**
     * For Android: closes the activity.
     */
    fun finish()

    /**
     * Reporting : to export to CSV
     */
    fun generateCSVReport()

    fun generateXLSXReport(xlsxReportPath: String)

    companion object {

        val VIEW_NAME = "ReportAttendanceGroupedByThresholds"
    }

}
